src/lib/check-in.ts

```diff
@@ -18,45 +89,37 @@ export function makeFetchCheckinDataAttempts(
     };
   };
 
-  return SwClient.loadJsonPage<Response>({
-    url: SwClient.withSuffix(
-      'mobile-air-operations/v1/mobile-air-operations/page/check-in/',
-      reservation
-    ),
-    method: SwClient.Method.GET,
-    headers,
-    retry: {
-      // We'll try a total of 80 times.
-      // This should give us 5 seconds of checkin tries before checkin time and 15 seconds after.
-      limit: 79,
-      methods: [SwClient.Method.POST],
-      statusCodes: [400, HttpStatus.NOT_FOUND],
-      calculateDelay: state => {
-        const nowTimestamp = Luxon.DateTime.now().toLocaleString(
-          Luxon.DateTime.DATETIME_FULL_WITH_SECONDS
-        );
-        if (logger) {
-          // It's normal to see many of these messages before we get the good response. We try
-          // many times because checkin does not become available at exactly the time expected.
-          logger.log(
-            'Failed on attempt %d of %d at %s with error:',
-            state.attemptCount,
-            state.retryOptions.limit,
-            nowTimestamp,
-            state.error.response?.body
-          );
-        }
-
-        // cancel retry when limit is hit
-        if (state.attemptCount > state.retryOptions.limit) {
-          return 0;
-        }
-
-        // retry every 250 milliseconds
-        return 250;
-      }
+  let response;
+  try {
+    response = await SwClient.loadJsonPage<Response>({
+      url: SwClient.withSuffix(
+        'mobile-air-operations/v1/mobile-air-operations/page/check-in/',
+        input.reservation
+      ),
+      method: SwClient.Method.GET,
+      headers: input.headers
+    });
+  } catch (error) {
+    if (input.logger) {
+      const nowTimestamp = Luxon.DateTime.now().toLocaleString(
+        Luxon.DateTime.DATETIME_FULL_WITH_SECONDS
+      );
+      // It's normal to see many of these messages before we get the good response. We try
+      // many times because the airline's API has a quirk of not making checkin available at
+      // exactly the time expected.
+      input.logger.log(
+        'Failed to fetch checkin data on attempt %d of %d at %s with error:',
+        input.attempt,
+        input.attemptLimit,
+        nowTimestamp,
+        error
```

Comment:
It'll be nice to keep the logging the same as before so we log the response body (which has all the error info as well).

Reply:
Thanks for testing this PR! I've updated this to log out `util.inspect(error)`. I'm not sure what the structure of this error looks like, but this should just print everything. If you still have a flight ready for checkin, maybe you can try to check in with this change and see if the full error yields any more information about why we're getting 404 responses.

Reply:
I'll book a flight today to test it :P. I have another flight upcoming this saturday that's I'll actually be going on, so hoping to test if it'll beat humans haha.

Reply:
I made another change, make sure you pull again